services/flows-operator/src/FlowOperator.js

```diff
@@ -111,19 +117,29 @@ class FlowOperator {
                         await channel.deleteExchange(exchange);
                     }
                 }
-                const queues =  await this._queueCreator.makeQueuesForTheTask(flowModel);
-                for (let node of flowModel.nodes) {
+                const queues =  await this._queueCreator.makeQueuesForTheFlow(flow);
+                for (let node of flow.nodes) {
                     this._logger.trace({flow: flow.metadata.name, node: node.id}, 'Going to create flow node');
-                    await this._deployNode(flowModel, node, queues);
+                    await this._deployNode(flow, node, queues);
                 }
             } else {
                 this._logger.trace({name: flow.metadata.name}, 'Nothing changed. Ensure nodes and queues exists');
+
+                if (!jobsIndex[flowId]) {
+                    const flowSecret = await this._getFlowSecret(flow);
```

Comment:
One optimization may be implemented here:
we may fetch all secrets in once and build hash-table for them instead of asking this from k8s api for every flow